pkg/controller/nodelifecycle/node_lifecycle_controller_test.go

```diff
@@ -167,561 +166,685 @@ func newNodeLifecycleControllerFromClient(
+						CreationTimestamp: fakeNow,
+						Labels: map[string]string{
+							v1.LabelTopologyRegion:          "region1",
+							v1.LabelTopologyZone:            "zone1",
+							v1.LabelFailureDomainBetaRegion: "region1",
+							v1.LabelFailureDomainBetaZone:   "zone1",
 						},
 					},
 				},
-				Clientset: fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{*testutil.NewPod("pod0", "node0")}}),
-			},
-			daemonSets: nil,
-			newNodeStatus: v1.NodeStatus{
-				Conditions: []v1.NodeCondition{
-					{
-						Type:   v1.NodeReady,
-						Status: v1.ConditionFalse,
-						// Node status has just been updated, and is NotReady for 10min.
-						LastHeartbeatTime:  metav1.Date(2015, 1, 1, 12, 9, 0, 0, time.UTC),
-						LastTransitionTime: metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "node3",
+						CreationTimestamp: fakeNow,
+						Labels: map[string]string{
+							v1.LabelTopologyRegion:          "region1",
+							v1.LabelTopologyZone:            "zone1",
+							v1.LabelFailureDomainBetaRegion: "region1",
+							v1.LabelFailureDomainBetaZone:   "zone1",
+						},
 					},
 				},
 			},
-			secondNodeNewStatus: healthyNodeNewStatus,
-			expectedEvictPods:   true,
-			description:         "Node created long time ago, and kubelet posted NotReady for a short period of time.",
+			updatedNodeStatuses: []v1.NodeStatus{
+				unhealthyNodeNewStatus,
+				unhealthyNodeNewStatus,
+				unhealthyNodeNewStatus,
+				healthyNodeNewStatus,
+			},
+			expectedInitialStates: map[string]ZoneState{
+				// we've not received any status for the nodes yet
+				// so the controller assumes the zones is fully disrupted
+				testutil.CreateZoneID("region1", "zone1"): stateFullDisruption,
+			},
+			expectedFollowingStates: map[string]ZoneState{
+				testutil.CreateZoneID("region1", "zone1"): statePartialDisruption,
+			},
 		},
-		// Pod is ds-managed, and kubelet posted NotReady for a long period of time.
-		{
-			fakeNodeHandler: &testutil.FakeNodeHandler{
-				Existing: []*v1.Node{
-					{
-						ObjectMeta: metav1.ObjectMeta{
-							Name:              "node0",
-							CreationTimestamp: metav1.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
-							Labels: map[string]string{
-								v1.LabelTopologyRegion:          "region1",
-								v1.LabelTopologyZone:            "zone1",
-								v1.LabelFailureDomainBetaRegion: "region1",
-								v1.LabelFailureDomainBetaZone:   "zone1",
-							},
-						},
-						Status: v1.NodeStatus{
-							Conditions: []v1.NodeCondition{
-								{
-									Type:               v1.NodeReady,
-									Status:             v1.ConditionFalse,
-									LastHeartbeatTime:  metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-									LastTransitionTime: metav1.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-								},
-							},
+		"Partial Disruption: one Node failed leading to the number of healthy Nodes to exceed the configured threshold": {
```

Comment:
qq: is the default threshold 50%?

Reply:
it is computed dynamically based on `nc.unhealthyZoneThreshold`

Reply:
All of the tests are using the `testUnhealthyThreshold` constant which is set to `0.55`.